### ISSUE TITLE
Parsec 1.5 snap build not starting

### ISSUE THREAD

**Issue by touilleMan** [Scille/parsec-cloud#954]
On ubuntu 19.10

```
$ parsec
QStandardPaths: XDG_RUNTIME_DIR points to non-existing path '/run/user/1000/snap.parsec', please create it with 0700 permissions.
QStandardPaths: XDG_RUNTIME_DIR points to non-existing path '/run/user/1000/snap.parsec', please create it with 0700 permissions.
/snap/parsec/167/usr/bin/python3: symbol lookup error: /usr/lib/x86_64-linux-gnu/libfontconfig.so.1: undefined symbol: FT_Done_MM_Var`
````

**Comment by vxgmichel** [comment]
Cannot reproduce on `Linux Mint 19.3 Tricia` (ubuntu 18.03)

**Comment by touilleMan** [comment]
On parsec startup, libfontconfig is loaded (I couldn't determine which library requires libfontconfig, I guess it's from CPython given we would have a stacktrace if it was loaded from python code)

so libfontconfig version is not the same between ubuntu 18.04 (version 2.12.6, https://packages.ubuntu.com/fr/bionic/libfontconfig1) and 19.10 (version 2.13.1, https://packages.ubuntu.com/fr/eoan/libfontconfig1)

Sad thing is those version are not compatible we build the snap with unbuntu 18.04, so running the snap on ubuntu 19.10 endup with:

```
/snap/parsec/167/usr/bin/python3: symbol lookup error: /usr/lib/x86_64-linux-gnu/libfontconfig.so.1: undefined symbol: FT_Done_MM_Var
```

The solution is to add libfontconfig into the snapcraft.yml included packages list. With this the application starts fine but we still have error logs:

```
Fontconfig warning: "/etc/fonts/fonts.conf", line 5: unknown element "its:rules"
Fontconfig warning: "/etc/fonts/fonts.conf", line 6: unknown element "its:translateRule"
Fontconfig error: "/etc/fonts/fonts.conf", line 6: invalid attribute 'translate'
Fontconfig error: "/etc/fonts/fonts.conf", line 6: invalid attribute 'selector'
Fontconfig error: "/etc/fonts/fonts.conf", line 7: invalid attribute 'xmlns:its'
Fontconfig error: "/etc/fonts/fonts.conf", line 7: invalid attribute 'version'
Fontconfig warning: "/etc/fonts/fonts.conf", line 9: unknown element "description"
Fontconfig warning: "/etc/fonts/conf.d/10-hinting-slight.conf", line 4: unknown element "its:rules"
Fontconfig warning: "/etc/fonts/conf.d/10-hinting-slight.conf", line 5: unknown element "its:translateRule"
Fontconfig error: "/etc/fonts/conf.d/10-hinting-slight.conf", line 5: invalid attribute 'translate'
Fontconfig error: "/etc/fonts/conf.d/10-hinting-slight.conf", line 5: invalid attribute 'selector'
Fontconfig error: "/etc/fonts/conf.d/10-hinting-slight.conf", line 6: invalid attribute 'xmlns:its'
Fontconfig error: "/etc/fonts/conf.d/10-hinting-slight.conf", line 6: invalid attribute 'version'
Fontconfig warning: "/etc/fonts/conf.d/10-hinting-slight.conf", line 8: unknown element "description"
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 4: unknown element "its:rules"
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 5: unknown element "its:translateRule"
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 5: invalid attribute 'translate'
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 5: invalid attribute 'selector'
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 6: invalid attribute 'xmlns:its'
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 6: invalid attribute 'version'
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 8: unknown element "description"
Fontconfig warning: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 4: unknown element "its:rules"
Fontconfig warning: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 5: unknown element "its:translateRule"
Fontconfig error: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 5: invalid attribute 'translate'
Fontconfig error: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 5: invalid attribute 'selector'
Fontconfig error: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 6: invalid attribute 'xmlns:its'
Fontconfig error: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 6: invalid attribute 'version'
Fontconfig warning: "/etc/fonts/conf.d/11-lcdfilter-default.conf", line 8: unknown element "description"
[...]
Fontconfig error: Cannot load config file from /etc/fonts/fonts.conf
```

Configuration file format is not compatible between versions 2.12 and 2.13

Correct configurate files are in `/snap/parsec/current/etc/fonts/` but I couldn't figure out how to tell libfontconfig to use them (I tried without success to use `FC_CONFIG_FILE` and `FC_CONFIG_DIR` env vars)

There is a bug on the libfontconfig bugtracker: https://bugs.freedesktop.org/show_bug.cgi?id=105818#c4
tl;dr: they don't want to do forward compatibility so it's a snap issue

In the end we currently have a quickfix https://github.com/Scille/parsec-build/commit/4518bcdc9e7ad4fb800922d79320ad7d39a406f2

A better solution would be to use the qt5 part config provided by the snap team:
https://github.com/ubuntu/snapcraft-desktop-helpers/blob/af0979e1e810732f75c39dbf85f8b6c32e3991e4/snapcraft.yaml#L242-L265

The thing is we have to configure PyQt to use this Qt library instead of the once shipped within the pypi package

It seems pyqtbundle is able to do the trick...

**Comment by c4ffein** [comment]
Fixed
